### ISSUE TITLE
Add API for extracting data from nodes

### ISSUE THREAD

**Issue by kpodsiad** [VirtusLab/scala-yaml#67]
### Discussed in https://github.com/VirtusLab/scala-yaml/discussions/56

<div type='discussions-op-text'>

<sup>Originally posted by **pikinier20** August 19, 2021</sup>
- Add an API to retrieve a value of a field with desired name

Examples:
```scala
case class Example(a: Int, b: String)
val yamlNode: Node = Node.MappingNode(
Node.KeyValueNode(Node.ScalarNode("a"), Node.ScalarNode(2)),
Node.KeyValueNode(Node.ScalarNode("b"), Node.ScalarNode("asd")
)
for {
a <- yamlNode.get("a").as[Int]
b <- yamlNode.get("b").as[String]
} yield Example(a, b)
```
</div>

**Comment by brianmowen** [comment]
@kpodsiad I'd like to use this library at work and want to help maintain and keep it up to date. Is this ticket a good place to start? Are you open to contributions?

**Comment by kpodsiad** [comment]
> Are you open to contributions?

Yes, you're more than welcome to contribute! One disclaimer though, this project is bit stale, I think there's nothing surprising in saying that there are just better things to focus on after work than YAML. Most probably some parts of the code could be written in a better way and might be unclear as there isn't many documentation. That being said, have you considered using https://github.com/circe/circe-yaml? If you expect solid piece of software this might be a better idea. It uses SnakaYaml to parse yaml files.

However, if I didn't scare you off and you're still interested, I can try to find some time to work on this library and help you with it.

**Comment by lbialy** [comment]
@brianmowen still interested in this issue? 🤔
